### PR TITLE
balancerd: correctly send passwords

### PR DESCRIPTION
**Pull request by mjibson**
Both balancerd and envd were sending password requests, causing clients to disconnect. Balancerd is changed to intercept the password request from envd and send what it has.

Add tests for the frontegg and static resolvers so they do not regress.

### Motivation

  * This PR fixes a previously unreported bug.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by jkosh44 — APPROVED** [review]
I don't have a ton of context into this, so I'm not entirely sure what the issue is, but the code itself LGTM.

File: src/balancerd/src/lib.rs [hunk 1]
```diff
@@ -325,11 +327,33 @@ impl PgwireBalancer {
...
-        if let Some(password) = password {
+        mz_stream.write_all(&buf).await?;
+        let client_stream = conn.inner_mut();
+
+        // Read a single backend message, which may be a password request. Send ours if so.
+        // Otherwise start shuffling bytes. message type (len 1, 'R') + message len (len 4, 8_i32) +
+        // auth type (len 4, 3_i32).
+        let mut maybe_auth_frame = [0; 1 + 4 + 4];
+        let nread = netio::read_exact_or_eof(&mut mz_stream, &mut maybe_auth_frame).await?;
+        const AUTH_PASSWORD_CLEARTEXT: [u8; 9] = [b'R', 0, 0, 0, 8, 0, 0, 0, 3];
```
[the remark this answers is not in the document]
Comment (mjibson) [reply src/balancerd/src/lib.rs]: Added.

**Comment by mjibson** [comment]
I described the issue in the PR message, not the commit message:

> Both balancerd and envd were sending password requests, causing clients to disconnect. Balancerd is changed to intercept the password request from envd and send what it has.

**Comment by nrainer-materialize** [comment]
The [nightly build](https://buildkite.com/materialize/nightlies/builds/4845) looks good (the failures are expected).

**Pull request merged**